parser/src/main/java/me/vertretungsplan/parser/LegionBoardParser.java

```diff
@@ -0,0 +1,197 @@
+		substitutionSchedule.setClasses(getAllClasses());
+		substitutionSchedule.setTeachers(getAllTeachers());
+		final JSONArray changes = getChanges();
+		final JSONArray courses = getCourses();
+		final JSONArray teachers = getTeachers();
+
+		parseLegionBoard(substitutionSchedule, changes, courses, teachers);
+		return substitutionSchedule;
+	}
+
+	/**
+	 * Returns authentication key as shown
+	 * <a href="https://gitlab.com/legionboard/heart/blob/master/doc/README.md">in the documentation</a>.
+	 */
+	private String getAuthenticationKey(Credential credential) {
+		final UserPasswordCredential userPasswordCredential = (UserPasswordCredential) credential;
+		final String username = userPasswordCredential.getUsername();
+		final String password = userPasswordCredential.getPassword();
+		return DigestUtils.sha256Hex(username.toLowerCase() + "//" + password);
+	}
+
+	/**
+	 * Returns a JSONArray with all changes from now to in one week.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/changes/list.md">List changes</a>
+	 */
+	private JSONArray getChanges() throws IOException, JSONException {
+		// Date (or alias of date) when the changes start
+		final String startBy = "now";
+		// Date (or alias of date) when the changes end
+		final String endBy = "i1w";
+
+		final String url = api + "/changes?startBy=" + startBy + "&endBy=" + endBy + "&k=" + getAuthenticationKey(getCredential());
+		return getJSONArray(url);
+	}
+
+	/**
+	 * Returns a JSONArray with all courses.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/courses/list.md">List courses</a>
+	 */
+	private JSONArray getCourses() throws IOException, JSONException {
+		final String url = api + "/courses?k=" + getAuthenticationKey(getCredential());
+		try {
+			return getJSONArray(url);
+		} catch (HttpResponseException e) {
+			// TODO: Implement
+			final boolean is404 = true;
+			if (is404) {
+				return null;
+			}
+			throw e;
+		}
+	}
+
+	/**
+	 * Returns a JSONArray with all teachers.
+	 * More information: <a href="https://gitlab.com/legionboard/heart/blob/master/doc/teachers/list.md">List teachers</a>
+	 */
+	private JSONArray getTeachers() throws IOException, JSONException {
+		final String url = api + "/teachers?k=" + getAuthenticationKey(getCredential());
+		try {
+			return getJSONArray(url);
+		} catch (HttpResponseException e) {
+			// TODO: Implement
+			final boolean is404 = true;
+			if (is404) {
+				return null;
+			}
+			throw e;
+		}
+	}
+
+	private JSONArray getJSONArray(String url) throws IOException, JSONException, HttpResponseException {
+		return new JSONArray(httpGet(url, "UTF-8"));
+	}
+
+	private void parseLegionBoard(SubstitutionSchedule substitutionSchedule, JSONArray changes, JSONArray courses, JSONArray teachers) throws IOException, JSONException {
+		if (changes == null) {
+			throw new IOException("LegionBoard has no changes.");
```

Comment:
Exception wird nicht mehr geworfen.
Standardmäßig werden in LegionBoard Eye (Webseite) die Änderungen von 
heute (Alias now) bis in einer Woche (Alias i1w/in 1 week) abgerufen. 
Wie sieht es bei Dir normal aus?